### PR TITLE
Some CMake fixes for Python bindings.

### PR DESCRIPTION
**Pull request by bilke**
See commit messages for details.

Related to https://gitlab.opengeosys.org/bilke/ogs/-/commits/mfront-python.

### CODE REVIEW

**Comment by thelfer** [comment]
Hi @bilke, would you comment on the change in the detection of the python interpreter ?
The current version tries to ensure that the `python` libraries are consistent with the interpreter. In your version, this check is removed. I don't remember the details, but the current version was done on purpose and does not seem to be a problem on all configurations tested so far.

**Comment by bilke** [comment]
Hi Thomas,

the CMake docs state:

> If calling both find_package(PythonInterp) and find_package(PythonLibs), call find_package(PythonInterp) first to get the currently active Python version by default with a consistent version of PYTHON_LIBRARIES.

And with my fix I got consistent behaviour per default (not specifying something Python related) and also with specifying a specific python version with e.g. `-DPYTHON_EXECUTABLE=python3.11`. The latter did not work before as `PythonLibs` picked up the default Python (e.g. 3.10) and the `PythonInterp` did then complain that a different version (the one I provided via the CMake variable, e.g. 3.11) was found than requested.

**Comment by thelfer** [comment]
@bilke Things changed over time concerning python detection in cmake. Both `PythonInterp` and `PythonLibs` are now deprecated and I am waiting to upgrade the the new `Python` module. Anyway, many scripts that we use relies on the definition of `PYTHON_ADDITIONAL_VERSIONS` to select a proper `python` version. Wouldn't that work for you ?

**Comment by bilke** [comment]
Ah thanks! I was not aware of that variable. I will let you know if that works...

**Comment by bilke** [comment]
Yes it seems to work! I will update this PR tomorrow and remove the python commit. I will also close the other PR.. Thanks!

**Comment by bilke** [comment]
I have removed the Python commit.

Regarding the test properties commit: for me I got a CMake error that `set_tests_properties()` was not called with right number of arguments. I guess the environment variables did not form a CMake list before.

File: bindings/python/tests/CMakeLists.txt [hunk 1]
```diff
@@ -34,9 +34,9 @@ function(test_python_bindings test)
 	CONFIGURATION ${conf})
       set_tests_properties(${test}_${conf}_py
     	PROPERTIES ENVIRONMENT
-        PYTHONPATH="${PROJECT_BINARY_DIR}/bindings/python/tests/;$ENV{PYTHONPATH}"
-        MGIS_TEST_BEHAVIOURS_LIBRARY="$<TARGET_FILE:BehaviourTest>"
-        MGIS_TEST_MODELS_LIBRARY="$<TARGET_FILE:ModelTest>")
+        "PYTHONPATH=${PROJECT_BINARY_DIR}/bindings/python/tests/;$ENV{PYTHONPATH}"
```
Comment (thelfer) [inline bindings/python/tests/CMakeLists.txt:37]: @bilke Could you comment on why you did this change. Is that what is broken now ?

Comment (bilke) [reply bindings/python/tests/CMakeLists.txt:37]: Without this change I get this error:

```
CMake Error at /cache/cpm/mgis/d91b741daaf78d4ef414125d6c96d30479c88d53/bindings/python/tests/CMakeLists.txt:35 (set_tests_properties):
  set_tests_properties called with incorrect number of arguments.
Call Stack (most recent call first):
  /cache/cpm/mgis/d91b741daaf78d4ef414125d6c96d30479c88d53/bindings/python/tests/CMakeLists.txt:53 (test_python_bindings)
```

I think as the value of `ENVIRONMENT` should be a list the problem is that `PYTHONPATH="${PROJECT_BINARY_DIR}/bindings/python/tests/;$ENV{PYTHONPATH}"` is not identified as a list item.

Comment (thelfer) [reply bindings/python/tests/CMakeLists.txt:37]: @bilke. Could you give information about your system ?

Comment (bilke) [reply bindings/python/tests/CMakeLists.txt]: @thelfer I think I got the issue corrected now: there was a semicolon used as a path separator, which confused CMake (as `;` is also the list separator in CMake) and should be `:` instead.

Comment (thelfer) [reply bindings/python/tests/CMakeLists.txt]: @bilke Nice. I believe that the `;` separator comes from `Windows`.  I need to check that. But for the moment, I merge the PR. Thanks again !

**Pull request merged**